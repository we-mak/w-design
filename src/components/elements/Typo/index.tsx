import React from "react";
import styled from "styled-components";
import { color } from "styled-system";
import { TypoProps } from "./types";
import { CustomTag } from "../../../index";

const WTypo = ({ tag, appearance, children, ...rest }: TypoProps) => (
  <CustomTag customHTMLtag={tag ? tag : appearance} {...rest}>
    {children}
  </CustomTag>
);

const Typo = styled(WTypo)`
  font-weight: 500;
  font-family: ${(props: any) => props.theme.fonts.medium};
  ${color};
  font-size: ${(props: any) => {
    const { fontSizes } = props.theme;

    switch (props.appearance) {
      case "h1":
        return fontSizes[6];
      case "h2":
        return fontSizes[5];
      case "h3":
        return fontSizes[4];
      case "h4":
        return fontSizes[3];
      case "h5":
        return fontSizes[2];
      case "h6":
        return fontSizes[1];
      default:
        return fontSizes[1];
    }
  }};
`;

Typo.displayName = "Typo";

export default Typo;
