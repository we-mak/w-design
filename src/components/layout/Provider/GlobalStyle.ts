import { createGlobalStyle } from "styled-components";
import theme from "../../../common/styleUtils/theme";
const colors = theme.colors;

/*! normalize.css v8.0.1 | MIT License | github.com/necolas/normalize.css */
export const GlobalStyle = createGlobalStyle`
html {
  line-height: 1.15;
  -webkit-text-size-adjust: 100%;
}
body {
  margin: 0;
}
main {
  display: block;
}
h1 {
  font-size: 2em;
  margin: 0.67em 0;
}
hr {
  box-sizing: content-box;
  height: 0;
  overflow: visible;
}
pre {
  font-family: monospace, monospace;
  font-size: 1em;
}
a {
  background-color: transparent;
}
abbr[title] {
  border-bottom: none;
  text-decoration: underline;
  text-decoration: underline dotted;
}
b,
strong {
  font-weight: bolder;
}
code,
kbd,
samp {
  font-family: monospace, monospace;
  font-size: 1em;
}
small {
  font-size: 80%;
}
sub,
sup {
  font-size: 75%;
  line-height: 0;
  position: relative;
  vertical-align: baseline;
}
sub {
  bottom: -0.25em;
}
sup {
  top: -0.5em;
}
img {
  border-style: none;
}
button,
input,
optgroup,
select,
textarea {
  font-family: inherit;
  font-size: 100%;
  line-height: 1.15;
  margin: 0;
}
button,
input {
  overflow: visible;
}
button,
select {
  text-transform: none;
}
[type="button"],
[type="reset"],
[type="submit"],
button {
  -webkit-appearance: button;
}
[type="button"]::-moz-focus-inner,
[type="reset"]::-moz-focus-inner,
[type="submit"]::-moz-focus-inner,
button::-moz-focus-inner {
  border-style: none;
  padding: 0;
}
[type="button"]:-moz-focusring,
[type="reset"]:-moz-focusring,
[type="submit"]:-moz-focusring,
button:-moz-focusring {
  outline: 1px dotted ButtonText;
}
fieldset {
  padding: 0.35em 0.75em 0.625em;
}
legend {
  box-sizing: border-box;
  color: inherit;
  display: table;
  max-width: 100%;
  padding: 0;
  white-space: normal;
}
progress {
  vertical-align: baseline;
}
textarea {
  overflow: auto;
}
[type="checkbox"],
[type="radio"] {
  box-sizing: border-box;
  padding: 0;
}
[type="number"]::-webkit-inner-spin-button,
[type="number"]::-webkit-outer-spin-button {
  height: auto;
}
[type="search"] {
  -webkit-appearance: textfield;
  outline-offset: -2px;
}
[type="search"]::-webkit-search-decoration {
  -webkit-appearance: none;
}
::-webkit-file-upload-button {
  -webkit-appearance: button;
  font: inherit;
}
details {
  display: block;
}
summary {
  display: list-item;
}
template {
  display: none;
}
[hidden] {
  display: none;
}

*,
*::before,
*::after {
  box-sizing: inherit;
}

html {
  box-sizing: border-box;
  font-size: 20px;
  line-height: 1.5;
  -webkit-tap-highlight-color: transparent;
}

body {
  font-size: .8rem;
  overflow-x: hidden;
  text-rendering: optimizeLegibility;
}


h1 {
    font-size: 1.6rem;
  }

  h2 {
    font-size: 1.4rem;
  }

  h3 {
    font-size: 1.2rem;
  }

  h4 {
    font-size: 1rem;
  }
  h5 {
    font-size: 0.9rem;
  }

  h6 {
    font-size: 0.8rem;
  }

  h1,
  h2,
  h3,
  h4,
  h5,
  h6 {
    font-weight: 700;
    line-height: 1.25;
    margin-bottom: 0.5em;
    margin-top: 0;
    font-family: ${(props: any) => props.theme.fonts.medium};
  }

  h1,
  h2,
  h3,
  h4 {
    color: ${colors.N50};
  }

  h5,
  h6 {
    color: ${colors.N60};
  }

  p {
    color: ${colors.N70};
    line-height: 1.5;
    margin-bottom: 0.5em;
    margin-top: 0;
    font-family: ${(props: any) => props.theme.fonts.normal};
  }

  a {
    font-family: ${(props: any) => props.theme.fonts.normal};
    color: ${colors.B50};
    outline: none;
    text-decoration: none;

    &:hover {
      color: ${colors.B30};
    }

    &:focus,
    &:hover,
    &:active {
      text-decoration: underline;
    }
    &:focus,
    &:active {
      color: ${colors.B60};
    }
    &:focus {
      text-decoration: none;
    }
  }

  b,
  strong {
    font-weight: bolder;
  }

  small {
    font-size: 80%;
    font-weight: 400;
    color: ${colors.N30};
  }

  blockquote {
    margin-left: 0;
    padding: 0.4rem 0.8rem;
    color: ${colors.N30};
    p:last-child {
      margin-bottom: 0;
    }
  }

  table {
    border-collapse: collapse;
    border-spacing: 0;
    text-align: left;
    width: 100%;
  }
  th {
    border-bottom: 0.05rem solid ${props => props.theme.colors["N5"]};
    padding: 0.6rem 0.4rem;
    border-bottom-width: 0.1rem;
  }
  td {
    border-bottom: 0.05rem solid ${props => props.theme.colors["N2"]};
    padding: 0.6rem 0.4rem;
  }
`;
