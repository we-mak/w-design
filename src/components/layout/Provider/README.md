Theme provider is based on [styled-components](https://www.styled-components.com/) and making theme with [styled-system](http://jxnblk.com/styled-system/)

Provider provides theme for your layout, you can supply your theme by insert your theme object by `theme` props
Default theme

```js static
defaultTheme = {
  breakpoint: ["24em", "32em", "48em", "64em", "80em"],
  fonts:
    '-apple-system, BlinkMacSystemFont, Helvetica, "Helvetica Neue", system-ui, sans-serif',
  fontSizes: [
    "0.7rem",
    "0.8rem",
    "0.9rem",
    "1rem",
    "1.2rem",
    "1.4rem",
    "1.6rem",
    "1.8rem",
    "2rem"
  ],
  fontWeights = { normal: 400, strong: 500, bold: 700, xbold: 900 },
  lineHeights = [1, 1.125, 1.25, 1.5],
  letterSpacings = { normal: "normal", caps: "0.25em" },
  radii = ["0rem", "0.1rem", "0.2rem", "0.3rem", "0.4rem"],
  space = [0, 4, 8, 16, 32, 64, 128], // space is used for margin and padding scales
  shadows = [
    "none",
    `0 .2rem .5rem ${hex2Rgba(colors.N50, 0.5)}`,
    `inset 0 0 0 1px rgba(87, 85, 217, .2), 0 0 4px rgba(87, 85, 217, .2)`
  ],
  colors: {...} // see colors
};
```

```js
<Provider>
  <h1>Default </h1>
  <p>
    Lorem ipsum
    <span>
      <a href="#">dolor </a>
    </span>
    sip amet
  </p>
</Provider>
```

```js
<Provider
  theme={{
    fonts: "Roboto, Helvetica, sans-serif"
  }}
>
  <h1>Roboto</h1>
  <p>
    Lorem ipsum
    <span>
      <a href="#">dolor </a>
    </span>
    sip amet
  </p>
</Provider>
```
