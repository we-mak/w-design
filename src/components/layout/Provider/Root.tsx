import styled from "../../../utils/styled";
import theme from "../../../common/theme";

const colors = theme.colors;

export const Root = styled.div`
  font-family: ${(props: any) => props.theme.fonts};

  h1 {
    font-size: 2rem;
  }

  h2 {
    font-size: 1.6rem;
  }

  h3 {
    font-size: 1.4rem;
  }

  h4 {
    font-size: 1.2rem;
  }
  h5 {
    font-size: 1rem;
  }

  h6 {
    font-size: 0.8rem;
  }

  h1,
  h2,
  h3,
  h4,
  h5,
  h6 {
    font-weight: 600;
    line-height: 1.25;
    margin-bottom: 0.5em;
    margin-top: 0;
  }

  h1,
  h2,
  h3,
  h4 {
    color: ${colors.N90};
  }

  h5,
  h6 {
    color: ${colors.N80};
  }

  p {
    color: ${colors.N70};
    line-height: 1.5;
    margin-bottom: 0.5em;
    margin-top: 0;
  }

  a {
    color: ${colors.B50};
    outline: none;
    text-decoration: none;

    &:hover {
      color: ${colors.B30};
    }

    &:focus,
    &:hover,
    &:active {
      text-decoration: underline;
    }
    &:focus,
    &:active {
      color: ${colors.B60};
    }
    &:focus {
      text-decoration: none;
    }
  }

  b,
  strong {
    font-weight: bolder;
  }

  code,
  kbd,
  pre,
  samp {
    font-family: "Menlo, Courier, monospace";
  }

  code {
    color: ${colors.R20};
    background: ${colors.N1};
    padding: 0.1rem 0.15rem;
    line-height: 1.2;
  }

  small {
    font-size: 80%;
    font-weight: 400;
    color: ${colors.N30};
  }

  blockquote {
    margin-left: 0;
    padding: 0.4rem 0.8rem;
    color: ${colors.N30};
    p:last-child {
      margin-bottom: 0;
    }
  }
`;

Root.displayName = "Root";

export default Root;
