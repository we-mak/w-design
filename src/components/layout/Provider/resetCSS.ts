import { colors } from "../../../common/colors";

const ResetCss = `
*,
*::before,
*::after {
  box-sizing: inherit;
}

html {
  box-sizing: border-box;
  font-size: 20px;
  line-height: 1.5;
  -webkit-tap-highlight-color: transparent;
}

body {
  background: ${colors.WHITE};
  color: ${colors.N60};
  font-family: -apple-system, BlinkMacSystemFont, Helvetica, "Helvetica Neue", system-ui, sans-serif;
  font-size: 0.8rem;
  overflow-x: hidden;
  text-rendering: optimizeLegibility;
}

a {
  font-family: Helvetica, "Helvetica Neue", system-ui, sans-serif;
}
`;

export default ResetCss;
