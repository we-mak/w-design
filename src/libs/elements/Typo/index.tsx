import React from "react";
import styled from "styled-components";
import { color, margin, fontWeight, fontFamily } from "styled-system";
import { TypoProps } from "./types";
import CustomTag from "../CustomTag";

const WTypo = ({ tag, appearance, children, ...rest }: TypoProps) => (
  <CustomTag tag={tag ? tag : appearance} {...rest}>
    {children}
  </CustomTag>
);

const Typo = styled(WTypo)`
  font-weight: 700;
  font-family: ${(props: TypoProps) => props.theme.fonts.medium};
  ${color};
  ${margin};
  ${fontWeight};
  ${fontFamily};
  font-size: ${(props: TypoProps) => {
    const { fontSizes } = props.theme;

    switch (props.appearance) {
      case "h1":
        return `${fontSizes[6]} !important`;
      case "h2":
        return `${fontSizes[5]} !important`;
      case "h3":
        return `${fontSizes[4]} !important`;
      case "h4":
        return `${fontSizes[3]} !important`;
      case "h5":
        return `${fontSizes[2]} !important`;
      case "h6":
        return `${fontSizes[1]} !important`;
      default:
        return `${fontSizes[1]} !important`;
    }
  }};
`;

Typo.displayName = "Typo";

export default Typo;
