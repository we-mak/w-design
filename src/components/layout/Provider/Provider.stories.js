import React from "react";
import { storiesOf } from "@storybook/react";
import Provider from "./Provider";

storiesOf("components/Provider", module).addWithJSX("default Provider", () => (
  <Provider>
    <h1>Default </h1>
    <p>
      Lorem ipsum
      <span>
        <a href="#"> dolor </a>
      </span>{" "}
      sip amet
    </p>
  </Provider>
));
