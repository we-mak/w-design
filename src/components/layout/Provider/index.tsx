import * as React from "react";
import { ThemeProvider } from "styled-components";
import defaultTheme from "../../../common/theme";
import { ResetCss } from "./ResetCss";
import Root from "./Root";

export interface Props extends React.HTMLProps<HTMLDivElement> {
  theme?: any;
}

class Provider extends React.Component<Props> {
  stylesheet: any;

  static displayName = "WProvider";

  public static defaultProps: Props = {
    theme: {}
  };

  componentDidMount() {
    this.stylesheet = document.createElement("style");
    this.stylesheet.type = "text/css";
    // this.stylesheet.setAttribute("rel", "preload");
    // this.stylesheet.setAttribute("as", "style");
    this.stylesheet.innerHTML = ResetCss;

    if (document && document.head) {
      document.head.appendChild(this.stylesheet);
    }
  }

  componentWillUnmount() {
    if (this.stylesheet && document && document.head) {
      document.head.removeChild(this.stylesheet);
      delete this.stylesheet;
    }
  }

  render() {
    const { theme, ...props } = this.props;
    return (
      <ThemeProvider theme={{ ...defaultTheme, ...theme }}>
        <Root {...props as any} />
      </ThemeProvider>
    );
  }
}

export default Provider;
