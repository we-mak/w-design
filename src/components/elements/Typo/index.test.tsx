import * as React from "react";
import { render, cleanup } from "react-testing-library";
import "jest-styled-components";
import Typo from "./index";
import Provider from "../../layout/Provider";

afterEach(cleanup);

describe("<Typo/>", () => {
  it("should render correct with heading style", () => {
    const { container } = render(
      <Provider>
        <Typo>Hello</Typo>
        <Typo appearance="h1">Hello</Typo>
        <Typo appearance="h2">Hello</Typo>
        <Typo appearance="h3">Hello</Typo>
        <Typo appearance="h4">Hello</Typo>
        <Typo appearance="h5">Hello</Typo>
        <Typo appearance="h6">Hello</Typo>
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });

  it("should render div with H1 style", () => {
    const { container } = render(
      <Provider>
        <Typo appearance="h1" tag="div">
          Hello
        </Typo>
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
