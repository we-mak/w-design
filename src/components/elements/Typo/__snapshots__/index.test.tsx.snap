// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Typo/> should render correct with heading style 1`] = `
.c1 {
  font-weight: 500;
  font-family: Helvetica,Arial,sans-serif;
  font-size: 1.6rem;
}

.c2 {
  font-weight: 500;
  font-family: Helvetica,Arial,sans-serif;
  font-size: 1.4rem;
}

.c3 {
  font-weight: 500;
  font-family: Helvetica,Arial,sans-serif;
  font-size: 1.2rem;
}

.c4 {
  font-weight: 500;
  font-family: Helvetica,Arial,sans-serif;
  font-size: 1rem;
}

.c5 {
  font-weight: 500;
  font-family: Helvetica,Arial,sans-serif;
  font-size: 0.9rem;
}

.c6 {
  font-weight: 500;
  font-family: Helvetica,Arial,sans-serif;
  font-size: 0.8rem;
}

.c0 {
  font-family: Helvetica,Arial,sans-serif;
}

.c0 h1 {
  font-size: 2rem;
}

.c0 h2 {
  font-size: 1.6rem;
}

.c0 h3 {
  font-size: 1.4rem;
}

.c0 h4 {
  font-size: 1.2rem;
}

.c0 h5 {
  font-size: 1rem;
}

.c0 h6 {
  font-size: 0.8rem;
}

.c0 h1,
.c0 h2,
.c0 h3,
.c0 h4,
.c0 h5,
.c0 h6 {
  font-weight: 700;
  line-height: 1.25;
  margin-bottom: 0.5em;
  margin-top: 0;
}

.c0 h1,
.c0 h2,
.c0 h3,
.c0 h4 {
  color: #394B59;
}

.c0 h5,
.c0 h6 {
  color: #293742;
}

.c0 p {
  color: #202B33;
  line-height: 1.5;
  margin-bottom: 0.5em;
  margin-top: 0;
}

.c0 a {
  color: #008CC0;
  outline: none;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 a:hover {
  color: #00A7E4;
}

.c0 a:focus,
.c0 a:hover,
.c0 a:active {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0 a:focus,
.c0 a:active {
  color: #007099;
}

.c0 a:focus {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 b,
.c0 strong {
  font-weight: bolder;
}

.c0 code,
.c0 kbd,
.c0 pre,
.c0 samp {
  font-family: "Menlo, Courier, monospace";
}

.c0 code {
  color: #F22828;
  background: #FBFBFB;
  padding: 0.1rem 0.15rem;
  line-height: 1.2;
}

.c0 small {
  font-size: 80%;
  font-weight: 400;
  color: #566E82;
}

.c0 blockquote {
  margin-left: 0;
  padding: 0.4rem 0.8rem;
  color: #566E82;
}

.c0 blockquote p:last-child {
  margin-bottom: 0;
}

<div>
  <div
    class="c0"
  >
    <h1
      class="c1"
    >
      Hello
    </h1>
    <h2
      class="c2"
    >
      Hello
    </h2>
    <h3
      class="c3"
    >
      Hello
    </h3>
    <h4
      class="c4"
    >
      Hello
    </h4>
    <h5
      class="c5"
    >
      Hello
    </h5>
    <h6
      class="c6"
    >
      Hello
    </h6>
  </div>
</div>
`;

exports[`<Typo/> should render div with H1 style 1`] = `
.c1 {
  font-weight: 500;
  font-family: Helvetica,Arial,sans-serif;
  font-size: 1.6rem;
}

.c0 {
  font-family: Helvetica,Arial,sans-serif;
}

.c0 h1 {
  font-size: 2rem;
}

.c0 h2 {
  font-size: 1.6rem;
}

.c0 h3 {
  font-size: 1.4rem;
}

.c0 h4 {
  font-size: 1.2rem;
}

.c0 h5 {
  font-size: 1rem;
}

.c0 h6 {
  font-size: 0.8rem;
}

.c0 h1,
.c0 h2,
.c0 h3,
.c0 h4,
.c0 h5,
.c0 h6 {
  font-weight: 700;
  line-height: 1.25;
  margin-bottom: 0.5em;
  margin-top: 0;
}

.c0 h1,
.c0 h2,
.c0 h3,
.c0 h4 {
  color: #394B59;
}

.c0 h5,
.c0 h6 {
  color: #293742;
}

.c0 p {
  color: #202B33;
  line-height: 1.5;
  margin-bottom: 0.5em;
  margin-top: 0;
}

.c0 a {
  color: #008CC0;
  outline: none;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 a:hover {
  color: #00A7E4;
}

.c0 a:focus,
.c0 a:hover,
.c0 a:active {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0 a:focus,
.c0 a:active {
  color: #007099;
}

.c0 a:focus {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 b,
.c0 strong {
  font-weight: bolder;
}

.c0 code,
.c0 kbd,
.c0 pre,
.c0 samp {
  font-family: "Menlo, Courier, monospace";
}

.c0 code {
  color: #F22828;
  background: #FBFBFB;
  padding: 0.1rem 0.15rem;
  line-height: 1.2;
}

.c0 small {
  font-size: 80%;
  font-weight: 400;
  color: #566E82;
}

.c0 blockquote {
  margin-left: 0;
  padding: 0.4rem 0.8rem;
  color: #566E82;
}

.c0 blockquote p:last-child {
  margin-bottom: 0;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      Hello
    </div>
  </div>
</div>
`;
